stackrox/stackrox
# chore(ui): Adjust some Td and Th elements for consistency

### Description

Prerequisite to add lint rules in pluginPatternFly.js file:
* td-dataLabel: Require that Td element has dataLabel prop for responsive cell content
* td-dataLabel-Th-text: Require that dataLabel (with string value) of Td corresponds to text (or screenReaderText) of Th

### Problem

Some inconsistencies prevent lint rules to ignore the minority of irrelevant elements.

### Analysis

Find in Files for elements implies some corresponding elements are not in same file:
* `<Th` has 429 results in 92 files
* `<Td` has 540 results in 95 files

Find in Files for props implies opportunities to replace abstract with concrete rendering. See **Residue** item 1.
* `dataLabel="` has 372 results in 78 files
* `dataLabel={` has 39 results in 15 files

First draft of td-dataLabel reports:
* 2 errors related to radio input in AccessControl folder. See **Residue** item 2.
* 3 errors in FlowsTable.tsx file that are fixed in #13193
* 5 errors in RiskAcceptance folder which is deleted in #13186

First draft of td-dataLabel-Th-text:
* 1 error in DelegatedRegistriesTable.tsx file
    `Th` element has screen reader text: Row deletion
    `Td` element has `dataLabel` prop: Delete row
* 1 error in VulnReportsPage.tsx file:
    `MyLastJobStatusTh` element has no text
    `Td` element has `dataLabel` prop: My last job status

### Solution

1. Edit DelegatedRegistriesTable.tsx file.
    * Replace inconsistent specific text with consistent generic **Rox action** text.
    * Follow example of WatchedImagesTable.tsx file.
2. Edit WatchedImageTable.tsx file.
    * Although specific **Remove watched image** text is consistent, replace with generic **Row action** text.
    More replacements:
    * NodeCves/Overview/CVEsTable.tsx
    * PlatformCves/Overview/CVEsTable.tsx
    * ImageOverviewTable.tsx
    * ImageVulnerabilitiesTable.tsx
    * WorkloadCVEOverviewTable.tsx
3. For `MyLastJobStatusTh` element has no text, refactor:
    * Rename MyJobStatusTh.tsx as JobStatusPopoverContent.tsx file.
    * Move HelpIconTh.tsx file from Containers to Components folder for reuse.
    * Replace `MyLastJobStatusTh` with `HelpIconTh` element that has `popoverContent` prop and **My last job status** as text for lint rule.
4. Replace some empty `Td` in `Thead` with `Th` that have screen reader text.
    * InitBundlesTable.tsx
    * ScanConfigTablesPage.tsx
    * ViolationsTablePanel.tsx
    * VulnReportsPage.tsx

### Residue

1. Replace abstract object.property idiom with concrete string in `Td` and `Th` elements.
2. For 2 errors related to radio input in AccessControl folder, use `select` prop with `variant='radio'` property.
3. Replace `actions` of empty `Td` with `isActionCell` prop and `ActionsColumn` child element.

### User-facing documentation


### Testing and quality


#### Automated testing


#### How I validated my change

1. `npm run lint` in ui/apps/platform
2. `npm run build` in ui/apps/platform and then compute branch - master for the following
    * `wc build/static/js/*.js`
        main.js `0 = 4619781 - 4619781`
        total `421 = 11745507 - 11745086`
    * `ls -al build/static/js/*.js | wc`
        files `0 = 180 - 180`
3. `npm run start` in ui/apps/platform

#### Manual testing

1. Visit /main/clusters/delegated-image-scanning, click **Specified registries**, and then enter a registry in the table.

    * Before changes, see inconsistent specific text and `dataLabel` at 1440px.
        ![DelegatedRegistriesTable_1440_generic](https://github.com/user-attachments/assets/cae017d6-69ea-4b33-b520-54cd7698e8bc)

    * After changes, see consistent generic text and `dataLabel` at 1440px.
        ![DelegatedRegistriesTable_1440_generic](https://github.com/user-attachments/assets/e8512fd1-87da-464b-b8b1-6cc9a696a1b8)

    * Before changes, see inconsistent specific text and `dataLabel` at 760px.
        ![DelegatedRegistriesTable_760_specific](https://github.com/user-attachments/assets/1e721191-0e09-4752-8ece-26416130923a)

    * After changes, see consistent generic text and `dataLabel` at 760px.
        ![DelegatedRegistriesTable_760_generic](https://github.com/user-attachments/assets/c63947b4-1578-433d-9be6-b5370d383cc3)

2. Visit /main/vulnerabilities/workload-cves and then click **Manage watched images**.

    * Before changes, see specific text and `dataLabel` at 1440px.
        ![WatchedImagesTable_1440px_specific](https://github.com/user-attachments/assets/e75fafe1-93fe-4dae-a879-405e5e5264df)

    * After changes, see generic text and `dataLabel` at 1440px.
        ![WatchedImagesTable_1440px_generic](https://github.com/user-attachments/assets/58b26895-6eb8-4222-8d4e-2230a5800f43)

    * Before changes, see specific text and `dataLabel` at 760px.
        ![WatchedImagesTable_760px_specific](https://github.com/user-attachments/assets/def27f4d-beef-473e-94a8-b0062f032f9e)

    * After changes, see generic text and `dataLabel` at 760px.
        ![WatchedImagesTable_760px_generic](https://github.com/user-attachments/assets/8b8115ec-552c-4038-a161-461610a61022)

3. Visit /main/vulnerabilities/workload-cves, click **Images** and then click to open kebab menu.

    * Before changes, see specific text of `Th` element at 1440px.
        ![ImageOverviewTable_1440px_specific](https://github.com/user-attachments/assets/68bc6663-76cf-4172-97e6-99d9c57fd8fe)

    * After changes, see generic text of `Th` element at 1440px.
        ![ImageOverviewTable_1440px_generic](https://github.com/user-attachments/assets/3449205f-13b0-4db6-bd0c-fd5bc58b69ab)

    * Independent of changes, usual kebab for row actions at 760px.
        ![ImageOverviewTable_760px](https://github.com/user-attachments/assets/b8f915fc-f103-4739-847a-0a2df596b1b2)

4. Visit /main/vulnerabilities/reports and add a report.

    * Before changes, see popover and markup from `Th` element at 1440px.
        ![VulnReportsPage_1440px_baseline](https://github.com/user-attachments/assets/ca80b103-a614-405e-ad63-a1d26032dcb5)

    * After changes, see same popover and markup from `Th` element at 1440px.
        ![VulnReportsPage_1440px_refactor](https://github.com/user-attachments/assets/40cbd277-f5f6-41f0-a2b3-e70c96c8594e)

    * Independent of changes, see consistent `dataLabel` prop at 760px.
        ![VulnReportsPage_760px](https://github.com/user-attachments/assets/12545021-f0b0-4dee-b0fa-48be24d7fd1b)
